### DISCUSSION TITLE
Navlist groups don't auto expand on active navlist items

### DISCUSSION THREAD

**Discussion by idealerror** [livewire/flux#128] [Feature Requests]
By default, `<flux:navlist.group>` are `:expanded="true"`.  

If set to `false` and a link within the group is active, the item is no longer visible as the group is not expanded.  

Can there be logic added to the navlist group to detect if a child item is active and automatically expand the group or is this too complicated and we should account for this ourselves on the group using `:expanded="(request()->is('/') || request()->is('/test'))"` or similar?

Separately, line 25 of navlist.group has a typo with double quotes:
```php
<div class="block space-y-[2px]"">
```

**Comment by joshhanley** [comment]
@idealerror thanks for reporting! I've converted this to a feature request for now, while it's not ideal it could be nice to have.

**Comment by ahosker** [comment]
@idealerror thanks for this. I couldn't get expanded true/false to work. I was missing the colon, but you also upgraded what I wanted. Thanks

```blade
<flux:navlist.group expandable heading="Tools" :expanded="(request()->is('tools/*'))">
@livewire('components.children', ['post'=> 3, 'title'=>'Tools', 'is_menu'=>true])
</flux:navlist.group>
```

**Comment by danito** [comment]
I have solved this by adding the currentRouteNames of all children, but will check your workaround.
```html
  <flux:navlist.group expandable :expanded="in_array(Route::currentRouteName(),['route1','route2'])" heading="My NavList Header" class="hidden lg:grid">
        <flux:navlist.item icon="home-modern" href="{{ route('route1') }}">Item 1</flux:navlist.item>
       <flux:navlist.item icon="document-check" href="{{ route('route2') }}">Item2</flux:navlist.item>
</flux:navlist.group>

```

**Comment by jpmorby** [comment]
I have spent literally hours trying to make this work on a multi group dropdown menu and was getting nowhere .. every "refresh" (in my otherwise single page application) caused the menus to reset ... and then I discovered @persist / @endpersist!

Wrap the nested foreach with that and job done!

**Comment by snapey** [comment]
@persist not working for me because the navbar is not a livewire component. Its part of the layout.app file

A compromise is to keep just the section open that the user is currently working in.

    <flux:navlist.group expandable :expanded="Route::is('rotas.*')" heading="Staff Rotas" class="">

Tip, `Route::is()` accepts any number of route names eg `Route::is('rotas.*', 'my-shifts')`

**Comment by PrinceMinky** [comment]
Hello, 

The below snippet works for me and my app. 

```
                <flux:navlist.group :heading="__('User Management')" :expandable="true" :expanded="request()->routeIs('admin.user.*')">
                    <flux:navlist.item icon="users" :href="route('admin.user.index')" :current="request()->routeIs('admin.user.index')" wire:navigate>{{ __('All Users') }}</flux:navlist.item>
                    <flux:navlist.item icon="user-group" :href="route('admin.user.role')" :current="request()->routeIs('admin.user.role')" wire:navigate>{{ __('Roles') }}</flux:navlist.item>
                    <flux:navlist.item icon="shield-exclamation" :href="route('admin.user.permission')" :current="request()->routeIs('admin.user.permission')" wire:navigate>{{ __('Permissions') }}</flux:navlist.item>
                </flux:navlist.group>
```

**Reply by jeffchown** [reply]
@PrinceMinky I'm pretty sure you can simplify your code by removing all of the `:current=...` attributes as the `flux:navlist.item`s will automatically set the appropriate item as current since each `:current` checks for a route exactly match your `:href`s

**Reply by PrinceMinky** [reply]
> @PrinceMinky I'm pretty sure you can simplify your code by removing all of the `:current=...` attributes as the `flux:navlist.item`s will automatically set the appropriate item as current since each `:current` checks for a route exactly match your `:href`s

I remember seeing a screencast, or somebody somewhere, mention that is the case and I didn't apply it. Thank you!

**Reply by jeffchown** [reply]
You're welcome, @PrinceMinky !
